web/src/js/__tests__/components/ValueEditor/ValueEditorSpec.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react'
+import renderer from 'react-test-renderer'
+import TestUtils from 'react-dom/test-utils'
+import ValueEditor from '../../../components/ValueEditor/ValueEditor'
+import { Key } from '../../../utils'
+
+describe('ValueEditor Component', () => {
+
+    let mockFn = jest.fn()
+    it ('should render correctly', () => {
+        let valueEditor = renderer.create(
+            <ValueEditor content="foo" onDone={mockFn}/>
+        ),
+            tree = valueEditor.toJSON()
+        expect(tree).toMatchSnapshot()
+    })
+
+    let valueEditor = TestUtils.renderIntoDocument(
+        <ValueEditor content="<script>foo</script>" onDone={mockFn}/>
+    )
+    it('should handle this.blur', () => {
+        valueEditor.input.blur = jest.fn()
+        valueEditor.blur()
+        expect(valueEditor.input.blur).toHaveBeenCalled()
+    })
+
+    it('should handle reset', () => {
+        valueEditor.reset()
+        expect(valueEditor.input.innerHTML).toEqual(
+            _.escape("<script>foo</script>")
```

Comment:
Make this `"&lt;script&gt;foo&lt;/script&gt;"` directly to avoid the dependency here?